### PR TITLE
Allow null maxIdle when merging map entries in a 3.11 cluster

### PR DESCRIPTION
**Pull request by mmedenjak**
WAN events do not carry the maxIdle value. The WAN event processing uses
map merge functionality and expects it to handle the scenario when
maxIdle is null. The fix allows for this case.

NOTE: 
Tests are in EE and they are currently `@Ignore`d until this fix is merged. 

NOTE 2:
Also, having such WAN specifics in map merge classes is not very good. I've been considering adding a `WanMapMergingEntryImpl` which would allow the `MapMergingEntryImpl` to continue evolving without any fear of breaking WAN compatibility. This can allow us better evolvability in the future but we still need the changes in `MapMergingEntryImpl`. The reason is as follows:
- the `WanMapMergingEntryImpl` would be constructed when we are processing WAN events but only if the cluster version is 3.11 or higher (see [here](https://github.com/hazelcast/hazelcast-enterprise/blob/master/hazelcast-enterprise/src/main/java/com/hazelcast/map/impl/EnterpriseMapReplicationSupportingService.java#L215)). For instance,
```
MapMergeTypes mergingEntry = nodeEngine.getClusterService().getClusterVersion().isGreaterOrEqual(Versions.V3_11)
                    ? createWanMergingEntry(nodeEngine.getSerializationService(), event.getEntryView()) 
                    : createMergingEntry(nodeEngine.getSerializationService(), event.getEntryView());
``` 
- let's say the cluster version is 3.10 and we are in the middle of RU. We then proceed to create the current `MapMergingEntryImpl`. After this, the cluster version can change to 3.11 before we actually serialize the entry. We again hit the same issue then - the old `MapMergingEntryImpl` will be constructed with a `maxIdle == null` and we get a NPE on serialization.
Bottom line - we can add a WAN specific merging entry and it might help us in the future. But we still need this check for now. I appreciate any input on the subject.

EE: https://github.com/hazelcast/hazelcast-enterprise/pull/2414

### CODE REVIEW

**Comment by mmedenjak** [comment]
Thanks for the review @tkountis @vbekiaris ! I'll introduce the WAN-specific merge class in a separate PR then.
